smartsim/entity/files.py

```diff
@@ -130,3 +130,95 @@ def _check_path(self, file_path):
         if path.isdir(full_path):
             return full_path
         raise SSConfigError(f"File or Directory {file_path} not found")
+
+
+class TaggedFilesHierarchy:
+    """TaggedFilesHierarchy maintains a list of files and a list of child
+    TaggedFilesHierarchies. The TaggedFilesHierarchy class is ment to be easily
+    converted into a directory file structure each with each dir containing a
+    copy of the specifed (possilby tagged) files specifed at each level of the
+    hierarchy via a simply depth first search over the hierarchy.
+    """
+
+    def __init__(self, base=""):
+        """Initialize a TaggedFilesHierarchy
+
+        :param base: pathlike string specifing the generated directory
+                     files are located
+        :type base: str, optional
+        """
+        self.base = base
+        self.files = set()
+        self.dirs = set()
+
+    @classmethod
+    def from_list_paths(cls, path_list, dir_contents_to_base=False):
+        """Given a list of absolute paths to files and dirs, create and return
+        a TaggedFilesHierarchy instance to representing the file hierarchy
+
+        :param path_list: list of absolute path like strings to tagged files
+                          or dirs containing tagged files
+        :type path_list: list of str
```

Comment:
We usually write `list[str]`, even though it is not a strict rule.

Reply:
resolved in https://github.com/CrayLabs/SmartSim/pull/119/commits/a7d354285401199f71581397fc4dd02f951de63d